UnityProject/Assets/Scripts/Systems/InGameEvents/InGameEventScripts/EventAsteroidDrift.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Managers;
+using Map;
+using Strings;
+using UnityEngine;
+
+namespace InGameEvents
+{
+	public class EventAsteroidDrift : EventScriptBase
+	{
+		private MatrixInfo asteroid = null;
+		private float velecoity = 250;
+
+		public override void OnEventStart()
+		{
+			asteroid =
+				MatrixManager.Instance.ActiveMatricesList.FindAll(x => x.Name.ToLower().Contains("asteroid")).PickRandom();
+
+			Debug.Log(asteroid == null);
+			if (asteroid == null)
+			{
+				return;
+			}
+			Debug.Log(asteroid.Name);
+
+			if (AnnounceEvent)
+			{
+				var text = "An asteroid has been detected drifting away from its position." +
+				           "All shuttles are advised to practice caution and slow down when navigating space.";
+
+				CentComm.MakeAnnouncement(ChatTemplates.CentcomAnnounce, text, CentComm.UpdateSound.NoSound);
+				_ = SoundManager.PlayNetworked(CommonSounds.Instance.AnnouncementAlert);
+			}
+
+			if (FakeEvent) return;
+
+			base.OnEventStart();
+		}
+
+		public override void OnEventStartTimed()
+		{
+			velecoity = Random.Range(0.45f, 0.95f);
+			UpdateManager.Add(CallbackType.FIXED_UPDATE, MoveShip);
```

Comment:
matrix stuff doesn't use fixed update

Reply:
but its physics? what else does it use?

Reply:
Update, but it has a more accurate Delta time

Reply:
Does our update manager make Update more consistent for physics?

Reply:
It has its own measurement for its own delta time call, so it exactly knows how long it's taken since the last update for that script
```
		ElapsedTimeSinceLastUpdate.Stop();
		float DeltaTimeSeconds = (float) ElapsedTimeSinceLastUpdate.Elapsed.TotalSeconds;
		ElapsedTimeSinceLastUpdate.Reset(); //TODO Editor pausing??
		ElapsedTimeSinceLastUpdate.Start();
```

Reply:
though somewhere if it's paused for too long it bugs out and go superfast for some reason, so I think there's a time * 
time somewhere

Reply:
updated